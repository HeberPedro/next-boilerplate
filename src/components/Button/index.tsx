import { ButtonHTMLAttributes } from 'react'
import * as S from './styles'

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode
}

const Button = ({ ...children }: ButtonProps) => <S.Wrapper {...children} />

export default Button
