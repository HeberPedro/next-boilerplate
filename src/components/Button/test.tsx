import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Button from '.'

describe('[COMPONENT]: Button', () => {
  it('It should render button', () => {
    const mockText = 'Test'

    const { container } = render(<Button>{mockText}</Button>)
    const button = screen.getByRole('button', { name: mockText })

    expect(button).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('It should run onClick function when button is pressed', () => {
    const mockOnClick = jest.fn()

    render(<Button onClick={mockOnClick()} />)
    const button = screen.getByRole('button')
    userEvent.click(button)

    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
