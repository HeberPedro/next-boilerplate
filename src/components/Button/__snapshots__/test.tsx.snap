// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT]: Button It should render button 1`] = `
<div>
  <button
    class=""
  >
    Test
  </button>
</div>
`;
